orangecontrib/explain/widgets/owice.py

```diff
@@ -0,0 +1,859 @@
+
+        else:
+            y = self.__y_individual[:, -1]
+
+        # eps is pixel size dependent
+        vb: ICEPlotViewBox = self.getViewBox()
+        _, px_height = vb.viewPixelSize()
+        mask = np.abs(y - pos.y()) < px_height * 5  # 5 px
+        return np.flatnonzero(mask).tolist()
+
+    def _help_event(self, event: QGraphicsSceneHelpEvent):
+        if self.__mean_line_item is None:
+            return False
+
+        pos = self.__mean_line_item.mapFromScene(event.scenePos())
+        indices = self._indices_at(pos)
+        text = self._get_tooltip(indices)
+        if text:
+            QToolTip.showText(event.screenPos(), text, widget=self)
+            return True
+        return False
+
+    def _get_tooltip(self, indices: List[int]) -> str:
+        text = "<hr/>".join(self.__instance_tooltip(self.__data, idx) for idx
+                            in indices[:self.MAX_POINTS_IN_TOOLTIP])
+        if len(indices) > self.MAX_POINTS_IN_TOOLTIP:
+            text = f"{len(indices)} instances<hr/>{text}<hr/>..."
+        return text
+
+    @staticmethod
+    def __instance_tooltip(data: Table, idx: int) -> str:
+        def show_part(_point_data, singular, plural, max_shown, _vars):
+            cols = [escape('{} = {}'.format(var.name, _point_data[var]))
+                    for var in _vars[:max_shown + 2]][:max_shown]
+            if not cols:
+                return ""
+            n_vars = len(_vars)
+            if n_vars > max_shown:
+                cols[-1] = "... and {} others".format(n_vars - max_shown + 1)
+            return \
+                "<b>{}</b>:<br/>".format(singular if n_vars < 2 else plural) \
+                + "<br/>".join(cols)
+
+        parts = (("Class", "Classes", 4, data.domain.class_vars),
+                 ("Meta", "Metas", 4, data.domain.metas),
+                 ("Feature", "Features", 10, data.domain.attributes))
+        return "<br/>".join(show_part(data[idx], *cols) for cols in parts)
+
+
+class OWICE(OWWidget, ConcurrentWidgetMixin):
+    name = "ICE"
+    description = "Dependence between a target and a feature of interest."
+    keywords = ["ICE", "PDP", "partial", "dependence"]
+    icon = "icons/ICE.svg"
+    priority = 130
+
+    class Inputs:
+        model = Input("Model", (SklModel, RandomForestModel))
+        data = Input("Data", Table)
+
+    class Outputs:
+        selected_data = Output("Selected Data", Table, default=True)
+        annotated_data = Output(ANNOTATED_DATA_SIGNAL_NAME, Table)
+
+    class Error(OWWidget.Error):
+        domain_transform_err = Msg("{}")
+        unknown_err = Msg("{}")
+        not_enough_data = Msg("At least two instances are needed.")
+        no_cont_features = Msg("At least one numeric feature is required.")
+
+    class Information(OWWidget.Information):
+        data_sampled = Msg("Data has been sampled.")
+
+    buttons_area_orientation = Qt.Vertical
+
+    settingsHandler = PerfectDomainContextHandler()
```

Comment:
Because one can select a target value. In case of removing the value from the dataset, the DomainContextHandler would still match the domains, which can result in an error:

E.g.: File (Iris) -> RF -> ICE (select Iris-virginica as the target class) -> Remove the last 50 instances from the iris.tab and repeat the procedure.